src/mrseq/scripts/spiral_flash.py

```diff
@@ -0,0 +1,374 @@
+    min_tr = (
+        pp.calc_duration(gz)  # rf pulse
+        + pp.calc_duration(gzr)  # slice rewinder
+        + pp.calc_duration(gx[0], gy[0])  # readout gradient
+        + pp.calc_duration(gz_spoil)  # gradient spoiler or readout-re-winder
+    )
+
+    # calculate repetition time delay (tr_delay)
+    current_min_tr = min_tr + te_delay
+    tr_delay = 0 if tr is None else round_to_raster(tr - current_min_tr, system.block_duration_raster)
+
+    if not tr_delay >= 0:
+        raise ValueError(f'TR must be larger than {current_min_tr * 1000:.2f} ms. Current value is {tr * 1000:.2f} ms.')
+
+    print(f'\nCurrent echo time = {(min_te + te_delay) * 1000:.2f} ms')
+    print(f'Current repetition time = {(current_min_tr + tr_delay) * 1000:.2f} ms')
+
+    # choose initial rf phase offset
+    rf_phase = 0
+    rf_inc = 0
+
+    # create header
+    if mrd_header_file:
+        hdr = create_header(
+            traj_type='other',
+            encoding_fov=Fov(x=fov_xy, y=fov_xy, z=slice_thickness),
+            recon_fov=Fov(x=fov_xy, y=fov_xy, z=slice_thickness),
+            encoding_matrix=MatrixSize(n_x=int(n_readout), n_y=int(n_readout), n_z=1),
+            recon_matrix=MatrixSize(n_x=n_readout, n_y=n_readout, n_z=1),
+            dwell_time=adc.dwell,
+            slice_limits=Limits(min=0, max=n_slices, center=0),
+            k1_limits=Limits(min=0, max=len(gx), center=0),
+            k2_limits=Limits(min=0, max=1, center=0),
+        )
+
+        # write header to file
+        prot = ismrmrd.Dataset(mrd_header_file, 'w')
+        prot.write_xml_header(hdr.toXML('utf-8'))
+
+    # obtain noise samples
+    seq.add_block(pp.make_label(label='LIN', type='SET', value=0), pp.make_label(label='SLC', type='SET', value=0))
+    seq.add_block(adc, pp.make_label(label='NOISE', type='SET', value=True))
+    seq.add_block(pp.make_label(label='NOISE', type='SET', value=False))
+    seq.add_block(pp.make_delay(system.rf_dead_time))
+
+    if mrd_header_file:
+        acq = ismrmrd.Acquisition()
+        acq.resize(trajectory_dimensions=2, number_of_samples=adc.num_samples)
+        prot.append_acquisition(acq)
+
+    for slice_ in range(n_slices):
+        for spiral_ in range(-n_dummy_excitations, len(gx)):
+            # calculate current phase_offset if rf_spoiling is activated
+            if rf_spoiling_phase_increment > 0:
+                rf.phase_offset = rf_phase / 180 * np.pi
+                adc.phase_offset = rf_phase / 180 * np.pi
+
+            # set frequency offset for current slice
+            rf.freq_offset = gz.amplitude * slice_thickness * (slice_ - (n_slices - 1) / 2)
+
+            # add slice selective excitation pulse
+            seq.add_block(rf, gz)
+            seq.add_block(gzr)
+
+            # update rf phase offset for the next excitation pulse
+            rf_inc = divmod(rf_inc + rf_spoiling_phase_increment, 360.0)[1]
+            rf_phase = divmod(rf_phase + rf_inc, 360.0)[1]
+
+            if te is not None:
```

Comment:
IMO, this should be changed to "if not te_delay > 0", because we it could happen that we choose te to be exactly min_te, which would result in a delay with 0 duration, no?